neqo-transport/src/path.rs

```diff
@@ -706,8 +706,12 @@ impl Path {
 
     /// Make a datagram.
     pub fn datagram<V: Into<Vec<u8>>>(&mut self, payload: V) -> Datagram {
+        // Make sure to use the TOS value from before calling EcnInfo::on_packet_sent, which may
+        // update the ECN state and can hence change it - this packet should still be sent
+        // with the current value.
+        let tos = self.tos();
         self.ecn_info.on_packet_sent();
```

Comment:
As an alternative, `on_packet_sent` could be renamed to e.g. `on_sending_packet` and changed to return the _to-be-used_ ecn mark for the _to-be-send_ packet.

I don't feel strongly about it.